### PR TITLE
Adding utils to ease our life with AI mailets

### PR DESCRIPTION
**Pull request by chibenwa**
- Factorize integer parsing from string for Mailets
 - Improve MimeMessageBuilder

### CODE REVIEW

File: mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java [hunk 1]
```diff
@@ -107,4 +110,38 @@ public static String normalizeSubject(String subj, String prefix) {
...
+    public static int getInitParameterAsStrictlyPositiveInteger(String condition) throws MessagingException {
+        return getInitParameterAsStrictlyPositiveInteger(condition, Optional.<String>absent());
+    }
+
+    public static int getInitParameterAsStrictlyPositiveInteger(String condition, Optional<String> defaultValue) throws MessagingException {
+        Optional<String> value = Optional.fromNullable(condition)
+            .or(defaultValue);
+
+        if (Strings.isNullOrEmpty(value.orNull())) {
+            throw new MessagingException(" is required. It should be a strictly positive integer");
```
Comment (quynhn) [inline mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java]: what is required?

File: mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java [hunk 2]
```diff
@@ -107,4 +110,38 @@ public static String normalizeSubject(String subj, String prefix) {
...
+
+        if (valueAsInt < 1) {
+            throw new MessagingException("Expecting condition to be a strictly positive integer. Got " + value.get());
+        }
+        return valueAsInt;
+    }
+
+    private static int tryParseInteger(String value) throws MessagingException {
+        try {
+            System.out.println(value);
```
Comment (quynhn) [inline mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java]: ?

File: mailet/base/src/test/java/org/apache/mailet/base/MailetUtilTest.java [hunk 3]
```diff
@@ -85,6 +92,87 @@ public void getInitParameterShouldReturnAbsentWhenNull() {
...
+        expectedException.expect(MessagingException.class);
+
+        MailetUtil.getInitParameterAsStrictlyPositiveInteger("invalid", DEFAULT_VALUE);
+    }
+
+    @Test
+    public void getInitParameterAsStrictlyPositiveIntegerWithDefaultValueShouldThrowOnNegativeNumber() throws Exception {
+        expectedException.expect(MessagingException.class);
+
+        MailetUtil.getInitParameterAsStrictlyPositiveInteger("-1");
```
Comment (quynhn) [inline mailet/base/src/test/java/org/apache/mailet/base/MailetUtilTest.java]: The name testing function said "WithDefaultValue..." Does it expect one more parameter value for default?

File: mailet/base/src/test/java/org/apache/mailet/base/MailetUtilTest.java [hunk 4]
```diff
@@ -85,6 +92,87 @@ public void getInitParameterShouldReturnAbsentWhenNull() {
...
+
+    @Test
+    public void getInitParameterAsStrictlyPositiveIntegerWithDefaultValueShouldThrowOnEmptyString() throws Exception {
+        expectedException.expect(MessagingException.class);
+
+        MailetUtil.getInitParameterAsStrictlyPositiveInteger("", DEFAULT_VALUE);
+    }
+
+    @Test
+    public void getInitParameterAsStrictlyPositiveIntegerWithDefaultValueShouldThrowOnNull() throws Exception {
```
Comment (quynhn) [inline mailet/base/src/test/java/org/apache/mailet/base/MailetUtilTest.java]: ShouldThrow? But it does not have any expected throw exception in test

File: mailet/base/src/test/java/org/apache/mailet/base/test/MimeMessageBuilder.java [hunk 5]
```diff
@@ -305,22 +325,21 @@ public MimeMessage build() throws MessagingException {
         Preconditions.checkState(!(text.isPresent() && content.isPresent()), "Can not get at the same time a text and a content");
         MimeMessage mimeMessage = new MimeMessage(Session.getInstance(new Properties()));
         if (text.isPresent()) {
-            BodyPart bodyPart = new MimeBodyPart();
-            bodyPart.setText(text.get());
-            mimeMessage.setContent(bodyPart, "text/plain");
+            mimeMessage.setContent(text.get(), "text/plain");
```
Comment (rouazana) [inline mailet/base/src/test/java/org/apache/mailet/base/test/MimeMessageBuilder.java]: I'm not sure this fix is fine. When you get a mail, you don't get it with a textBody, you get a BinaryBody with an InputStream you need to parse. So I think we are trying to mimic this behaviour.
Can you check please?

Comment (chibenwa) [inline mailet/base/src/test/java/org/apache/mailet/base/test/MimeMessageBuilder.java]: I don't understand your comment. This code perfectly instanciate a MimeMessage with a simple text body.

Here I define I want a text/plain body. When needed we will handle logic for binary bodies. Anyway, I believe it is out of the scope of what I am doing.

Comment (rouazana) [inline mailet/base/src/test/java/org/apache/mailet/base/test/MimeMessageBuilder.java]: ok

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**